packages/amplify_core/lib/src/category/amplify_push_notifications_category.dart

```diff
@@ -0,0 +1,92 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+part of amplify_interface;
+
+class PushNotificationsCategory
+    extends AmplifyCategory<PushNotificationsPluginInterface> {
+  @override
+  @nonVirtual
+  Category get category => Category.notificationsPush;
+
+  static final PushNotificationsCategory _pushNotificationsCategory =
+      PushNotificationsCategory._internal();
+
+  factory PushNotificationsCategory() {
+    return _pushNotificationsCategory;
+  }
+  PushNotificationsCategory._internal();
+
+  /// {@template amplify_core.amplify_push_notifications_category.get_permission_status}
+  /// Returns the current Push Notificaiton permission status.
+  ///
+  /// The status can be one of the following:
+  /// 1. notRequested - Android only status to indicate push permissions dialogue has not been requested
+  /// 2. shouldRequestWithRationale - show a rationale message on why the app needs to send push notifications
+  /// 3. granted - end user has granted the request for the app to send push notifications
+  /// 4. denied - end user has denied the request for the app to send push notifications
+  /// {@endtemplate}
+  Future<PushNotificationPermissionRequestStatus> getPermissionStatus() =>
+      defaultPlugin.getPermissionStatus();
+
+  /// {@template amplify_core.amplify_push_notifications_category.request_permissions}
+  /// Requests push notifications permissions with configurable options.
+  ///
+  /// Options include [alert], [badge] and [sound].
```

Comment:
This just describes the API which is redundant... burning question is what do they do?

```suggestion
  /// On iOS, these control the [granular permissions](... link to Apple Developer docs)
  /// On Android, ....
```